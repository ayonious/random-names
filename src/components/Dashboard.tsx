import * as React from "react";
import CountryReults from "./CountryResults";
import Select from "@material-ui/core/Select";
import MenuItem from "@material-ui/core/MenuItem";
import FormHelperText from "@material-ui/core/FormHelperText";
import FormControl from "@material-ui/core/FormControl";
import { database } from "../resources/Database";
const regions = Object.keys(database);

interface Props {}

interface State {
  selectedCountry: string;
  currentName: string;
}

export default class Dashboard extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.onSelectOption = this.onSelectOption.bind(this);
    this.state = { selectedCountry: "Germany", currentName: "" };
  }

  onSelectOption(event: any) {
    this.setState({ selectedCountry: event.target.value });
  }

  render() {
    const selectList = regions.map(region => (
      <MenuItem key={region} value={region}>
        {database[region].flag} {region}
      </MenuItem>
    ));
    const style = {
      display: "flex",
      width: 160,
      margin: "0 auto",
      alignContent: "center",
      justifyContent: "center"
    };

    return (
      <div>
        <FormControl style={style}>
          <Select
            value={this.state.selectedCountry}
            onChange={this.onSelectOption}
            autoWidth={false}
          >
            {selectList}
          </Select>
          <FormHelperText>Region</FormHelperText>
        </FormControl>
        <CountryReults country={this.state.selectedCountry} />
      </div>
    );
  }
}
