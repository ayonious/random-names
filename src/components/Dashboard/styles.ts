import styled from 'styled-components';
import { theme, siteMaxWidth } from '../../../config';
export const FormWrapper = styled.div``;

export const SelectItemWrapper = styled.div`
  width: ${siteMaxWidth}; ;
`;

export const SelectTitleWrapper = styled.div`
  width: ${siteMaxWidth};
  margin-left: 10px;
  margin-top: 5px;
  color: ${theme.fontColor.light};
`;

export const SelectWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-items: center;
  align-items: center;
`;

export const PageWrapper = styled.div`
  margin: 50px;
`;

export const CustomSelectStyles = {
  option: (provided: any, state: any) => ({
    ...provided,
    background: state.isSelected
      ? theme.primaryColor
      : state.isFocused
      ? theme.primaryLightColor
      : 'white',
  }),
};
