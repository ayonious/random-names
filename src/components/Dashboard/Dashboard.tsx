import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import React, { useState } from 'react';

import { database } from '../../resources/Database';
import { FormWrapper } from './styles';
import CountryReults from '../CountryResults/CountryResults';
import Greeting from '../Greeting/Greeting';

const regions = Object.keys(database);

interface Props {}

interface State {
  selectedCountry: string;
}
const Dashboard = () => {
  const [selectedCountry, changeSelectedCountry] = useState('Germany');

  const onSelectOption = (event: any) => {
    changeSelectedCountry(event.target.value);
  };

  const selectList = regions.map((region) => (
    <MenuItem key={region} value={region}>
      {database[region].flag} {region}
    </MenuItem>
  ));
  return (
    <div>
      <div>
        <Greeting />
      </div>
      <FormWrapper>
        <FormControl className="myFormClass">
          <Select
            value={selectedCountry}
            onChange={onSelectOption}
            autoWidth={false}
          >
            {selectList}
          </Select>
          <FormHelperText>Region</FormHelperText>
        </FormControl>
        <CountryReults country={selectedCountry} />
      </FormWrapper>
    </div>
  );
};

export default Dashboard;
