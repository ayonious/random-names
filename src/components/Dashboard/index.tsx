import React, { useState } from 'react';
import { SingleValue } from 'react-select';

import { database } from '../../resources/Database';
import CountryResults from '../CountryResults';
import GithubCornerPart from '../GithubCornerPart';
import Greeting from '../Greeting';
import SelectCountry, { SelectItem } from '../SelectCountry';
import { PageWrapper } from './styles';

const regions = Object.keys(database);

const Dashboard = () => {
  const [selectedCountry, changeSelectedCountry] = useState<string>('Germany');

  const onSelectOption = (event: SingleValue<SelectItem>) => {
    changeSelectedCountry(event?.value || 'Germany');
  };

  const selectList: SelectItem[] = regions.map((region) => ({
    label: `${database[region].flag} ${region}`,
    value: region,
  }));

  return (
    <PageWrapper>
      <Greeting />
      <SelectCountry selectList={selectList} onSelectOption={onSelectOption} />
      <CountryResults country={selectedCountry} />
      <GithubCornerPart />
    </PageWrapper>
  );
};

export default Dashboard;
