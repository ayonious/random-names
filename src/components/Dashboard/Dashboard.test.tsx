/**
 * @jest-environment jsdom
 */
import { expect } from '@jest/globals';
import { render } from '@testing-library/react';
import React from 'react';

import Dashboard from '.';

it('Dashboard renders', () => {
  const { container } = render(<Dashboard />);
  expect(container).toMatchSnapshot();
});
