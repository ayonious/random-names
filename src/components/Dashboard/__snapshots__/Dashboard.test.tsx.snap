// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders 1`] = `
<div>
  <div
    class="sc-iBdmCd hhuQxv"
  >
    <div
      class="sc-dmyDGi huzMHV"
    >
      <h1>
         Start Shuffling! 
      </h1>
    </div>
    <div
      class="sc-hLQRIN"
    >
      <div
        class="sc-gLLvby gKfwUz"
      >
        <div
          class="sc-eDLJxc gMGDNt"
        >
          <div
            class=" css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-13cymwt-control"
            >
              <div
                class=" css-1fdsijx-ValueContainer"
              >
                <div
                  class=" css-1dimb5e-singleValue"
                >
                  ðŸ‡©ðŸ‡ª Germany
                </div>
                <div
                  class=" css-qbdosj-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-2-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-1xc3v61-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="country list"
              type="hidden"
              value="Germany"
            />
          </div>
        </div>
        <div
          class="sc-jTQDnj epqAHh"
        >
          Region
        </div>
      </div>
      <div>
        <br />
        <div
          class="sc-beySbM jefaWJ"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiCard-root css-1vqzrnv-MuiPaper-root-MuiCard-root"
            data-testid="name-card"
          />
        </div>
        <br />
        <div>
          <div
            class="sc-guDLRT imulVP"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              data-testid="shuffle-button"
              tabindex="0"
              type="button"
            >
              Shuffle
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <a
      aria-label="Open GitHub project"
      class="github-corner"
      href="https://github.com/ayonious/random-names"
    >
      <svg
        height="80"
        style="position: absolute; top: 0px; fill: #fff; right: 0px;"
        viewBox="0 0 250 250"
        width="80"
      >
        <path
          class="octo-banner"
          d="M0 0l115 115h15l12 27 108 108V0z"
          fill="#5f32b3"
        />
        <path
          class="octo-arm"
          d="M128 109c-15-9-9-19-9-19 3-7 2-11 2-11-1-7 3-2 3-2 4 5 2 11 2 11-3 10 5 15 9 16"
          style="transform-origin: 130px 106px;"
        />
        <path
          class="octo-body"
          d="M115 115s4 2 5 0l14-14c3-2 6-3 8-3-8-11-15-24 2-41 5-5 10-7 16-7 1-2 3-7 12-11 0 0 5 3 7 16 4 2 8 5 12 9s7 8 9 12c14 3 17 7 17 7-4 8-9 11-11 11 0 6-2 11-7 16-16 16-30 10-41 2 0 3-1 7-5 11l-12 11c-1 1 1 5 1 5z"
        />
      </svg>
    </a>
  </div>
</div>
`;
